src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableNestedClasses5.java

```diff
@@ -0,0 +1,32 @@
+/*
+UnusedLocalVariable
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;
+
+public class InputUnusedLocalVariableNestedClasses5 {
+  int a = 12;
+
+  void foo() {
+    int a = 12; // violation, unused variable 'a'
+    int ab = 12; // violation, unused variable 'ab'
+
+    class abc {
+      Test a = new Test() {
+        void abc() {
+          System.out.println(a);
+          int abc = 10; // violation, unused variable 'abc'
```

Comment:
Make another test case like this, but rename this variable to the same as the outter one, `ab`.

Reply:
Done

Reply:
Is this what is line 31 is?

Reply:
Yes.
>Make another test case like this, but rename this variable to the same as the outer one, ab.

I rename the var to `ab` (the same as the outer one). Am I understanding this wrong...

Reply:
````
  int a = 12;

  void foo() {
    int a = 12;
    int ab = 12;

    class abc {
      Test a = new Test() {
        void abc() {
          System.out.println(a);
          int ab = 10;

          class def {
            Test abc = new Test() {
              void def() {
                System.out.println(abc);
              }
            };
          }
        }
      };
    }
```

And another one where `Test abc = new Test() {` is `Test ab = new Test() {`.

Reply:
@rnveach I see, so just want to double check. I need to make a new test file for this example?

Reply:
Do you mean like this
```java
public class InputUnusedLocalVariableNestedClasses5 {
  int a = 12;

  void foo() {
    int a = 12; // violation, unused variable 'a'
    int ab = 12; // violation, unused variable 'ab'

    class abc {
      Test a = new Test() {
        void abc() {
          System.out.println(a);
          int abc = 10; // violation, unused variable 'abc'

          class def {
            Test abc = new Test() {
              void def() {
                System.out.println(abc);
              }
            };
            Test ab = new Test() {
              void def() {
                System.out.println(ab);
              }
            };
          }
        }
      };
    }
  }
}
```

Reply:
Done, as described above.